Refactor tests

Flood1993 opened the pull request:
A bit of test refactoring. No test functionality should have been modified. Main improvements:
- Reuse the same `receive` function, instead of having it declared in multiple files.
- Refactorization of `test_agent_async_requests` through fixtures.
- Add more consistency in the naming of the tests (`agent.received`, `handler=receive`...)

Peque commented on osbrain/tests/test_agent_async_requests.py:
```diff
@@ -55,6 +87,29 @@ def receive(agent, response):
+                           log_name='log_history_warning',
+                           message='Received response for an unknown request!')
+
+
```
Why move the `test_unknown` position?

Flood1993 replied:
I moved it simply to have the test that use the same texture in order.

Peque commented on osbrain/tests/test_agent_async_requests.py:
```diff
@@ -2,39 +2,71 @@
 Test file for asynchronous requests.
 """
 import time
+import pytest
```
I'd use a blank line to separate built-in packages from external packages (i.e.: between `time` and `pytest`).

This, however, is not really your fault. It should be included in the style/flakes tests. We could add an issue and do it when we find some time. :joy:

Peque commented on osbrain/tests/common.py:
```diff
@@ -6,6 +6,10 @@
 
 
+def receive(agent, message, topic=None):
+    agent.received.append(message)
```
I like the idea of moving this function to `common`. We could, however, use a more explicit name? In example: `set_received`, `append_received` or something like that. `receive` seems a bit vague.

Maybe even `set_received_handler`, `append_received_handler`...

What you think?

Flood1993 replied:
Adding `_handler` is a bit too much for my taste, but I will change it to `append_received` 👍

Flood1993 commented:
I fixed the whitespace between the imports, renamed the function, extracted set_received as well and removed the position change of the test to clean the git diff. I will add a PR with that change.
